README.md

```diff
@@ -240,41 +183,100 @@ HTTP(port).Register("/found", [](Request r) {
 struct PennyInput {
   std::string op;
   std::vector<int> x;
+  std::string error;  // Not serialized.
   template <typename A> void serialize(A& ar) {
     ar(CEREAL_NVP(op), CEREAL_NVP(x));
-  }   
-  // TODO(dkorolev): Safe mode wrt parsing malformed JSON.
+  }
+  void FromInvalidJSON(const std::string& input_json) {
+    error = "JSON parse error: " + input_json;
+  }
 };
 
 // An output record that would be sent back as a JSON.
 struct PennyOutput {
+  std::string error;
   int result;
   template <typename A> void serialize(A& ar) {
-    ar(CEREAL_NVP(result));
-  }   
-};  
+    ar(CEREAL_NVP(error), CEREAL_NVP(result));
+  }
+}; 
 
 // Doing Penny-level arithmetics for fun and performance testing.
 HTTP(port).Register("/penny", [](Request r) {
-  // TODO(dkorolev): `r.body`, and its safe mode.
-  const auto input = JSONParse<PennyInput>(r.http.Body());
-  int result = 0;
-  if (input.op == "add") {
-    for (const auto v : input.x) {
-      result += v;
-    }
-  } else if (input.op == "mul") {
-    result = 1;
-    for (const auto v : input.x) {
-      result *= v;
+  const auto input = JSONParse<PennyInput>(r.body);
+  if (!input.error.empty()) {
+    r(PennyOutput{input.error, 0});
+  } else {
+    int result = 0;
+    if (input.op == "add") {
+      for (const auto v : input.x) {
+        result += v;
+      }
+    } else if (input.op == "mul") {
+      result = 1;
+      for (const auto v : input.x) {
+        result *= v;
+      }
     }
+    r(PennyOutput{"", result});
```

Comment:
Just added a test. On the other hand, we also need a "not enough parameters" case, give me a sec.